sphinxgallery/gen_rst.py

```diff
@@ -445,7 +445,7 @@ def generate_file_rst(fname, target_dir, src_dir, gallery_conf, plot_gallery):
     Returns the set of functions/classes imported in the example.
     """
     base_image_name = os.path.splitext(fname)[0]
-    image_fname = '%s_%%03d.png' % base_image_name
+    image_fname = 'sphx_glr_%s_%%03d.png' % base_image_name
```

Comment:
> If that's too hard, I guess we can live with this PR as a stop-gap solution/hack.

what is stop-gap solution?

> Stupid questions:
> 
>     where is the code that copies the images from doc/auto_examples/images to _build/html/_images ? Is that inside sphinx or inside sphinx-gallery?

This is a sphinx feature. All images are copied to `_build/html/_images` no folder structure is kept. This is the reason why sphinx renames images with repeated names preventing overwrites and the reason new images keep increasing in the build folder as it never gets cleared. Until the next `make clean`

>     where does the matplotlib plot directive stores its output images, is that _build/html/_images ?

I don't know if it has a temporary cached dir, but in the end the matplotlib plot directive gets overtaken by sphinx and it sends the final images to `_build/html/_images` with the same script name. There is when the conflict starts if one uses scripts from the gallery, so one image gets renamed and sphinx keeps track of the naming. But on subsequent `make html` images are cleared at some point by the hack in sphinx-gallery the images of the gallery are copied again and linked properly by sphinx, but ones of the plot directive don't get copied again or get deleted after being set, so the output html has lost images but sphinx does not complain because it is sure to have copied the image to `_build/html/_images`. It is a synchronization and name conflict problem.

>     what are the proper variable names in sphinx-gallery for both of these folders?

Sphinx-gallery creates recursively two folders inside the `gallery_dir` one for images `image_dir` and one inside it for the thumbnails `thumb_dir`, and this structure is keep for every sub-folder of the `examples_dir`. Sphinx-gallery links to this images locally from the rst files, and when generating the html it copies individually the image to `_build/html/_images`(renaming it if necessary) and establishes the correct html link. The directory `_build/html` is a sphinx variable found in `app.builder.outdir` and the `_images` sub-folder is pure managed by sphinx